### PR TITLE
Integer Palindrome Check

### PR DESCRIPTION
**Pull request by shrutivanw**
CS Fundamentals' assignment to check if the digits in the integer form a palindrome.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,9 @@
...
\ No newline at end of file
+# Integer Palindrome Check
+Check if the digits in a non-negtive integer form a palindrome.
+
+## Exercise
+* Design and implement a method that checks if the digits in the input positive integer forms a palindrome. The method returns true if the digits in the input form a palindrome. The method returns false otherwise.
+* For example, 12521 and 4554 are palindromes. 326725 is not a palindrome.
+* Share and explain the time and space complexities for your solution.
+* If you think of multiple approaches to solve the problem, implement the solution which minimizes space complexity. Explain the other approaches, and your decision in comments above the code.
+**Note**: Do not use Ruby provided functionality.
```
Comment (Hamled) [inline README.md]: We should have an extra blank line above this so that the note goes into its own paragraph.

File: lib/integer_palindrome_check.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Returns true if the input positive integer number forms a palindrome. Returns false otherwise.
+def is_palindrome(number)
+  puts "NOT IMPLEMENTED"
+  return true
```
Comment (Hamled) [inline lib/integer_palindrome_check.rb]: How would you feel about doing `raise NotImplementedError` here instead?

Best practice I think is generally to avoid `puts` statements and other things that will print to the console for code being tested, because that output kind of gets randomly interspersed with the test runner's output.

File: specs/integer_palindrome_check_spec.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
+require 'minitest/autorun'
+require 'minitest/reporters'
+require_relative '../lib/integer_palindrome_check'
+
+describe "integer palindrome check" do
+  describe "basic tests" do
+    describe "test 1" do
```
Comment (Hamled) [inline specs/integer_palindrome_check_spec.rb]: Same as the other repos, we should probably remove the `describe` blocks that are immediate parents of individual tests.

File: specs/integer_palindrome_check_spec.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    describe "test 3" do
+      it "not a palindrome" do
+        is_palindrome(1728371).must_equal false
+      end
+    end
+  end
+
+  # if the parameter is an object, check for nil
+  describe "nil object input test" do
+    it "nil object is not an integer" do
```
Comment (Hamled) [inline specs/integer_palindrome_check_spec.rb]: I could see this test going into the edge cases section.

**Review by Hamled — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (shrutivanw) [reply README.md]: Done.

File: lib/integer_palindrome_check.rb (hunk 2 above)
Comment (shrutivanw) [reply lib/integer_palindrome_check.rb]: Done.

File: specs/integer_palindrome_check_spec.rb (hunk 3 above)
Comment (shrutivanw) [reply specs/integer_palindrome_check_spec.rb]: done!

File: specs/integer_palindrome_check_spec.rb (hunk 4 above)
Comment (shrutivanw) [reply specs/integer_palindrome_check_spec.rb]: done!

**Comment by shrutivanw** [comment]
Thank you @Hamled Feedback incorporated.

**Review by Hamled — APPROVED** [review]
